File: commons-core/src/test/java/xyz/wagyourtail/commons/core/string/NameTypeTest.java
package xyz.wagyourtail.commons.core.string;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class NameTypeTest {

    @Test
    public void testConvertCamelCaseToKebabCase() {
        assertEquals("test-test", NameType.CAMEL_CASE.convert(NameType.KEBAB_CASE, "testTest"));
        assertEquals("test-test-test", NameType.CAMEL_CASE.convert(NameType.KEBAB_CASE, "testTestTest"));
        // check with initialism
        assertEquals("THE-test", NameType.CAMEL_CASE.convert(NameType.KEBAB_CASE, "THETest"));
        assertEquals("test-THE", NameType.CAMEL_CASE.convert(NameType.KEBAB_CASE, "testTHE"));
    }

    @Test
    public void testConvertPascalCaseToKebabCase() {
        assertEquals("test-test", NameType.PASCAL_CASE.convert(NameType.KEBAB_CASE, "TestTest"));
        assertEquals("test-test-test", NameType.PASCAL_CASE.convert(NameType.KEBAB_CASE, "TestTestTest"));
        // check with initialism
        assertEquals("THE-test", NameType.PASCAL_CASE.convert(NameType.KEBAB_CASE, "THETest"));
        assertEquals("test-THE", NameType.PASCAL_CASE.convert(NameType.KEBAB_CASE, "TestTHE"));
    }

    @Test
    public void testConvertKebabCaseToCamelCase() {
        assertEquals("testTest", NameType.KEBAB_CASE.convert(NameType.CAMEL_CASE, "test-test"));
        assertEquals("testTestTest", NameType.KEBAB_CASE.convert(NameType.CAMEL_CASE, "test-test-test"));
        // check with initialism
        assertEquals("THETest", NameType.KEBAB_CASE.convert(NameType.CAMEL_CASE, "THE-test"));
        assertEquals("testTHE", NameType.KEBAB_CASE.convert(NameType.CAMEL_CASE, "test-THE"));
    }

    @Test
    public void testConvertKebabCaseToPascalCase() {
        assertEquals("TestTest", NameType.KEBAB_CASE.convert(NameType.PASCAL_CASE, "test-test"));
        assertEquals("TestTestTest", NameType.KEBAB_CASE.convert(NameType.PASCAL_CASE, "test-test-test"));
        // check with initialism
        assertEquals("THETest", NameType.KEBAB_CASE.convert(NameType.PASCAL_CASE, "THE-test"));
        assertEquals("TestTHE", NameType.KEBAB_CASE.convert(NameType.PASCAL_CASE, "test-THE"));
    }

    @Test
    public void testConvertCamelCaseToSnakeCase() {
        assertEquals("test_test", NameType.CAMEL_CASE.convert(NameType.SNAKE_CASE, "testTest"));
        assertEquals("test_test_test", NameType.CAMEL_CASE.convert(NameType.SNAKE_CASE, "testTestTest"));
        // check with initialism
        assertEquals("THE_test", NameType.CAMEL_CASE.convert(NameType.SNAKE_CASE, "THETest"));
        assertEquals("test_THE", NameType.CAMEL_CASE.convert(NameType.SNAKE_CASE, "testTHE"));
    }

    @Test
    public void testConvertPascalCaseToSnakeCase() {
        assertEquals("test_test", NameType.PASCAL_CASE.convert(NameType.SNAKE_CASE, "TestTest"));
        assertEquals("test_test_test", NameType.PASCAL_CASE.convert(NameType.SNAKE_CASE, "TestTestTest"));
        // check with initialism
        assertEquals("THE_test", NameType.PASCAL_CASE.convert(NameType.SNAKE_CASE, "THETest"));
        assertEquals("test_THE", NameType.PASCAL_CASE.convert(NameType.SNAKE_CASE, "TestTHE"));
    }

    @Test
    public void testConvertSnakeCaseToCamelCase() {
        assertEquals("testTest", NameType.SNAKE_CASE.convert(NameType.CAMEL_CASE, "test_test"));
        assertEquals("testTestTest", NameType.SNAKE_CASE.convert(NameType.CAMEL_CASE, "test_test_test"));
        // check with initialism
        assertEquals("THETest", NameType.SNAKE_CASE.convert(NameType.CAMEL_CASE, "THE_test"));
        assertEquals("testTHE", NameType.SNAKE_CASE.convert(NameType.CAMEL_CASE, "test_THE"));
    }

    @Test
    public void testConvertSnakeCaseToPascalCase() {
        assertEquals("TestTest", NameType.SNAKE_CASE.convert(NameType.PASCAL_CASE, "test_test"));
        assertEquals("TestTestTest", NameType.SNAKE_CASE.convert(NameType.PASCAL_CASE, "test_test_test"));
        // check with initialism
        assertEquals("THETest", NameType.SNAKE_CASE.convert(NameType.PASCAL_CASE, "THE_test"));
        assertEquals("TestTHE", NameType.SNAKE_CASE.convert(NameType.PASCAL_CASE, "test_THE"));
    }

    @Test
    public void testConvertCamelCaseToPascalCase() {
        assertEquals("TestTest", NameType.CAMEL_CASE.convert(NameType.PASCAL_CASE, "testTest"));
        assertEquals("TestTestTest", NameType.CAMEL_CASE.convert(NameType.PASCAL_CASE, "testTestTest"));
        // check with initialism
        assertEquals("THETest", NameType.CAMEL_CASE.convert(NameType.PASCAL_CASE, "THETest"));
        assertEquals("TestTHE", NameType.CAMEL_CASE.convert(NameType.PASCAL_CASE, "testTHE"));
    }

    @Test
    public void testConvertPascalCaseToCamelCase() {
        assertEquals("testTest", NameType.PASCAL_CASE.convert(NameType.CAMEL_CASE, "TestTest"));
        assertEquals("testTestTest", NameType.PASCAL_CASE.convert(NameType.CAMEL_CASE, "TestTestTest"));
        // check with initialism
        assertEquals("THETest", NameType.PASCAL_CASE.convert(NameType.CAMEL_CASE, "THETest"));
        assertEquals("testTHE", NameType.PASCAL_CASE.convert(NameType.CAMEL_CASE, "TestTHE"));
    }

}

File: commons-kt-kt/src/commonTest/kotlin/xyz/wagyourtail/commonskt/test/string/NameTypeTest.kt
package xyz.wagyourtail.commonskt.test.string

import xyz.wagyourtail.commonskt.string.NameType
import xyz.wagyourtail.commonskt.string.convertNameType
import kotlin.test.Test
import kotlin.test.assertEquals

class NameTypeTest {

    @Test
    fun testConvertCamelCaseToKebabCase() {
        assertEquals("test-test", "testTest".convertNameType(NameType.CAMEL_CASE, NameType.KEBAB_CASE))
        assertEquals("test-test-test", "testTestTest".convertNameType(NameType.CAMEL_CASE, NameType.KEBAB_CASE))
        // check with initialism
        assertEquals("THE-test", "THETest".convertNameType(NameType.CAMEL_CASE, NameType.KEBAB_CASE))
        assertEquals("test-THE", "testTHE".convertNameType(NameType.CAMEL_CASE, NameType.KEBAB_CASE))
    }

    @Test
    fun testConvertPascalCaseToKebabCase() {
        assertEquals("test-test", "TestTest".convertNameType(NameType.PASCAL_CASE, NameType.KEBAB_CASE))
        assertEquals("test-test-test", "TestTestTest".convertNameType(NameType.PASCAL_CASE, NameType.KEBAB_CASE))
        // check with initialism
        assertEquals("THE-test", "THETest".convertNameType(NameType.PASCAL_CASE, NameType.KEBAB_CASE))
        assertEquals("test-THE", "TestTHE".convertNameType(NameType.PASCAL_CASE, NameType.KEBAB_CASE))
    }

    @Test
    fun testConvertKebabCaseToCamelCase() {
        assertEquals("testTest", "test-test".convertNameType(NameType.KEBAB_CASE, NameType.CAMEL_CASE))
        assertEquals("testTestTest", "test-test-test".convertNameType(NameType.KEBAB_CASE, NameType.CAMEL_CASE))
        // check with initialism
        assertEquals("THETest", "THE-test".convertNameType(NameType.KEBAB_CASE, NameType.CAMEL_CASE))
        assertEquals("testTHE", "test-THE".convertNameType(NameType.KEBAB_CASE, NameType.CAMEL_CASE))
    }

    @Test
    fun testConvertKebabCaseToPascalCase() {
        assertEquals("TestTest", "test-test".convertNameType(NameType.KEBAB_CASE, NameType.PASCAL_CASE))
        assertEquals("TestTestTest", "test-test-test".convertNameType(NameType.KEBAB_CASE, NameType.PASCAL_CASE))
        // check with initialism
        assertEquals("THETest", "THE-test".convertNameType(NameType.KEBAB_CASE, NameType.PASCAL_CASE))
        assertEquals("TestTHE", "test-THE".convertNameType(NameType.KEBAB_CASE, NameType.PASCAL_CASE))
    }

    @Test
    fun testConvertCamelCaseToSnakeCase() {
        assertEquals("test_test", "testTest".convertNameType(NameType.CAMEL_CASE, NameType.SNAKE_CASE))
        assertEquals("test_test_test", "testTestTest".convertNameType(NameType.CAMEL_CASE, NameType.SNAKE_CASE))
        // check with initialism
        assertEquals("THE_test", "THETest".convertNameType(NameType.CAMEL_CASE, NameType.SNAKE_CASE))
        assertEquals("test_THE", "testTHE".convertNameType(NameType.CAMEL_CASE, NameType.SNAKE_CASE))
    }

    @Test
    fun testConvertPascalCaseToSnakeCase() {
        assertEquals("test_test", "TestTest".convertNameType(NameType.PASCAL_CASE, NameType.SNAKE_CASE))
        assertEquals("test_test_test", "TestTestTest".convertNameType(NameType.PASCAL_CASE, NameType.SNAKE_CASE))
        // check with initialism
        assertEquals("THE_test", "THETest".convertNameType(NameType.PASCAL_CASE, NameType.SNAKE_CASE))
        assertEquals("test_THE", "TestTHE".convertNameType(NameType.PASCAL_CASE, NameType.SNAKE_CASE))
    }

    @Test
    fun testConvertSnakeCaseToCamelCase() {
        assertEquals("testTest", "test_test".convertNameType(NameType.SNAKE_CASE, NameType.CAMEL_CASE))
        assertEquals("testTestTest", "test_test_test".convertNameType(NameType.SNAKE_CASE, NameType.CAMEL_CASE))
        // check with initialism
        assertEquals("THETest", "THE_test".convertNameType(NameType.SNAKE_CASE, NameType.CAMEL_CASE))
        assertEquals("testTHE", "test_THE".convertNameType(NameType.SNAKE_CASE, NameType.CAMEL_CASE))
    }

    @Test
    fun testConvertSnakeCaseToPascalCase() {
        assertEquals("TestTest", "test_test".convertNameType(NameType.SNAKE_CASE, NameType.PASCAL_CASE))
        assertEquals("TestTestTest", "test_test_test".convertNameType(NameType.SNAKE_CASE, NameType.PASCAL_CASE))
        // check with initialism
        assertEquals("THETest", "THE_test".convertNameType(NameType.SNAKE_CASE, NameType.PASCAL_CASE))
        assertEquals("TestTHE", "test_THE".convertNameType(NameType.SNAKE_CASE, NameType.PASCAL_CASE))
    }

    @Test
    fun testConvertCamelCaseToPascalCase() {
        assertEquals("TestTest", "testTest".convertNameType(NameType.CAMEL_CASE, NameType.PASCAL_CASE))
        assertEquals("TestTestTest", "testTestTest".convertNameType(NameType.CAMEL_CASE, NameType.PASCAL_CASE))
        // check with initialism
        assertEquals("THETest", "THETest".convertNameType(NameType.CAMEL_CASE, NameType.PASCAL_CASE))
        assertEquals("TestTHE", "testTHE".convertNameType(NameType.CAMEL_CASE, NameType.PASCAL_CASE))
    }

    @Test
    fun testConvertPascalCaseToCamelCase() {
        assertEquals("testTest", "TestTest".convertNameType(NameType.PASCAL_CASE, NameType.CAMEL_CASE))
        assertEquals("testTestTest", "TestTestTest".convertNameType(NameType.PASCAL_CASE, NameType.CAMEL_CASE))
        // check with initialism
        assertEquals("THETest", "THETest".convertNameType(NameType.PASCAL_CASE, NameType.CAMEL_CASE))
        assertEquals("testTHE", "TestTHE".convertNameType(NameType.PASCAL_CASE, NameType.CAMEL_CASE))
    }

}